operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -0,0 +1,635 @@
+// is inserted, then the CEP is queued in any one of the available CES. CEPs are
+// inserted into CESs without any preference or any priority.
+type cesMgr struct {
+
+	// desiredCESs is used to map CESName to CESTracker[i.e. list of CEPs],
+	// as well as CEPName to CESName.
+	desiredCESs *CESToCEPMapping
+
+	// workqueue is used to sync CESs with the api-server. this will rate-limit the
+	// CES requests going to api-server, ensures a single CES will not be proccessed
+	// multiple times concurrently, and if CES is added multiple times before it
+	// can be processed, this will only be processed only once.
+	queue workqueue.RateLimitingInterface
+
+	// maxCEPsInCES is the maximum number of CiliumCoreEndpoint(s) packed in
+	// a CiliumEndpointSlice Resource.
+	maxCEPsInCES int
+}
+
+// cesManagerFcfs use cesMgr by design, it inherits all the methods from the base cesMgr and there is no
+// special handling required for cesManagerFcfs.
+// cesManagerFcfs indicates ciliumEndpoints are batched based on FirstComeFirtServe algorithm.
+// refer cesMgr comments for more information.
+type cesManagerFcfs struct {
+	cesMgr
+}
+
+// cesManagerIdentity is used to batch CEPs in CES based on CEP identity.
+type cesManagerIdentity struct {
+	cesMgr
+	// Mutex to protect cep insert/removal in ces and removedCEPs
+	identityLock lock.RWMutex
+	// CEP identity to cesTracker map
+	identityToCES map[int64][]*cesTracker
+	// reverse map of identityToCES i.e. cesName to CEP identity
+	cesToIdentity map[string]int64
+}
+
+// newCESManagerFcfs creates and initializes a new FirstComeFirstServe based CES
+// manager, in this mode CEPs are batched based on FirstComeFirtServe algorithm.
+func newCESManagerFcfs(workQueue workqueue.RateLimitingInterface, maxCEPsInCES int) cesManager {
+	return &cesManagerFcfs{
+		cesMgr{
+			desiredCESs:  newDesiredCESMap(),
+			queue:        workQueue,
+			maxCEPsInCES: maxCEPsInCES,
+		},
+	}
+}
+
+// newCESManagerIdentity creates and initializes a new Identity based manager.
+func newCESManagerIdentity(workQueue workqueue.RateLimitingInterface, maxCEPsInCES int) cesManager {
+	c := cesMgr{
+		desiredCESs:  newDesiredCESMap(),
+		queue:        workQueue,
+		maxCEPsInCES: maxCEPsInCES,
+	}
+	return &cesManagerIdentity{
+		cesMgr:        c,
+		identityToCES: make(map[int64][]*cesTracker),
+		cesToIdentity: make(map[string]int64),
+	}
+}
+
+// addCEPtoCES inserts the CEP in a CES, if the CEP already exists in a CES
+// it replaces with new CEP.
+func (c *cesMgr) addCEPtoCES(cep *cilium_v2.CoreCiliumEndpoint, cesName string) {
+	ces := c.desiredCESs.getCESTrackerOnly(cesName)
+	ces.backendMutex.Lock()
+	defer ces.backendMutex.Unlock()
+	// If cep already exists in ces, compare new cep with cached cep.
+	// Update only if there is any change.
+	log.WithFields(logrus.Fields{
+		"CEPName":        cep.Name,
+		"CESName":        ces.ces.GetName(),
+		"total CEPCount": len(ces.ces.Endpoints),
```

Comment:
Nit: logfields shouldn't have spaces. It doesn't look as nice when looking at the logs 

```suggestion
		"totalCEPCount": len(ces.ces.Endpoints),
```

Reply:
sure, Done.